examples_tests/20.Megatexture/main.cpp

```diff
@@ -76,359 +51,87 @@ layout (push_constant) uniform Block {
     float d;
     float Ni;
     uint extra; //flags copied from MTL metadata
+};
+layout (push_constant) uniform Block {
+    PCstruct params;
 } PC;
 #define _IRR_FRAG_PUSH_CONSTANTS_DEFINED_
 )";
-constexpr const char* GLSL_VT_FUNCTIONS =
+constexpr const char* GLSL_TEXTURE_SAMPLE_OVERRIDE =
 R"(
-#include <irr/builtin/glsl/texture_packer/definitions.glsl/6/6/7/8>
 
-vec3 unpackPageID(in uint pageID)
-{
-	// this is optimal, don't touch
-	uvec2 pageXY = uvec2(pageID,pageID>>ADDR_Y_SHIFT)&uvec2(ADDR_X_MASK,ADDR_Y_MASK);
-	return vec3(vec2(pageXY),float(pageID>>ADDR_LAYER_SHIFT));
-}
+mat2 dUV;//global
```

Comment:
why is it global??

you could have made your `irr_sample_*(coord)` functions as `irr_sample_*(coord,derivatives)`